### PR TITLE
Ta 134 delivery logs should persist on db

### PR DESCRIPTION
**Pull request by Alex90Jennings**

### CODE REVIEW

File: src/controllers/deliveryLog.js [hunk 1]
```diff
@@ -1,24 +1,32 @@
...
-          content: line.content
-        }
-      })
-    }
-  })
+  const deliveryLogToCreate = await DeliveryLog.fromJson(newDeliveryLogData)
+  const deliveryLog = await deliveryLogToCreate.save()
+
+  const deliveryLogLines = []
+  const postId = deliveryLog.id
```
Comment (dearshrewdwit) [inline src/controllers/deliveryLog.js:18]: 1. rename this variable `postId`  to `deliveryLogId` here and where it's used
2. Or delete it -> seems unnecessary to store this data in a variable

File: src/controllers/deliveryLog.js [hunk 2]
```diff
@@ -1,24 +1,32 @@
...
+
+  for (const line of lines) {
```
Comment (dearshrewdwit) [inline src/controllers/deliveryLog.js:20]: Use `map()` instead. Something like:

```js
const deliveryLogLines = lines.map(line => {
    const data = { content: line.content, logId: deliveryLog.id }
    return await DeliveryLogLines.fromJson(data)
})
```

You could then also save it during the loop:

```js
const deliveryLogLines = lines.map(line => {
    const data = { content: line.content, logId: deliveryLog.id }
    const logLineToCreate =  await DeliveryLogLines.fromJson(data)
    return await logLineToCreate.save()
})
```

File: src/domain/deliveryLog.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    return new DeliveryLog(
+      deliveryLog.lines,
+      deliveryLog.date,
+      deliveryLog.userId,
+      deliveryLog.cohortId,
+      deliveryLog.id
+    )
+  }
+
+  static async fromJson(json) {
```
Comment (dearshrewdwit) [inline src/domain/deliveryLog.js:14]: Is there a reason this is `async`?

File: src/domain/deliveryLogLines.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+export default class DeliveryLogLines {
+  static fromDb(deliveryLogLines) {
+    return new DeliveryLogLines(
+      deliveryLogLines.content,
+      deliveryLogLines.logId
+    )
+  }
+
+  static async fromJson(json) {
```
Comment (dearshrewdwit) [inline src/domain/deliveryLogLines.js:11]: Is there a reason this is `async` ?

**Review by dearshrewdwit — CHANGES_REQUESTED** [review]

File: src/domain/deliveryLogLines.js (hunk 4 above)
Comment (Alex90Jennings) [reply src/domain/deliveryLogLines.js:11]: used post.js and user.js as a template, that has this static async from json. Should I remove? Or keep it for consistency?

File: src/domain/deliveryLog.js (hunk 3 above)
Comment (Alex90Jennings) [reply src/domain/deliveryLog.js:14]: used post.js and user.js as a template, that has this static async from json. Should I remove? Or keep it for consistency?

File: src/controllers/deliveryLog.js (hunk 2 above)
Comment (dearshrewdwit) [reply src/controllers/deliveryLog.js:20]: Share your solution here first - if it's reasonable, yes let's map.

File: src/domain/deliveryLogLines.js (hunk 4 above)
Comment (dearshrewdwit) [reply src/domain/deliveryLogLines.js:11]: Noted - see above

File: src/controllers/deliveryLog.js (hunk 2 above)
Comment (dearshrewdwit) [reply src/controllers/deliveryLog.js:20]: > Should I pass this to a current member of the teacher admin team?

@Alex90Jennings Yes

File: src/controllers/deliveryLog.js (hunk 1 above)
Comment (PBara7a) [reply src/controllers/deliveryLog.js:18]: Removed postId variable.

File: src/controllers/deliveryLog.js (hunk 2 above)
Comment (PBara7a) [reply src/controllers/deliveryLog.js:20]: The implemmentation is now using a map as suggested.

File: src/domain/deliveryLog.js (hunk 3 above)
Comment (PBara7a) [reply src/domain/deliveryLog.js:14]: As @Alex90Jennings said, other domains are using the async keyword on their static fromJson methods. 
I did try to remove it and I could not store a new DeliveryLog without it.

Comment (dearshrewdwit) [reply src/domain/deliveryLog.js:14]: It's not necessary - we can remove the async qualifier from all domain object methods like this one that don't actually perform an async operation.

That work will be in a different PR though.

**Review by dearshrewdwit — APPROVED** [review]

**Pull request merged**